lib/internal/child_process.js

```diff
@@ -764,9 +764,13 @@ function _validateStdio(stdio, sync) {
         fd: i
       });
     } else if (typeof stdio === 'number' || typeof stdio.fd === 'number') {
```

Comment:
I wonder if it would just be simpler to break this up into two separate `else if`s at this point.